x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/MlAutoUpdateService.java

```diff
@@ -42,13 +45,17 @@ public MlAutoUpdateService(ThreadPool threadPool, List<UpdateAction> updateActio
 
     @Override
     public void clusterChanged(ClusterChangedEvent event) {
+        if (event.state().blocks().hasGlobalBlock(GatewayService.STATE_NOT_RECOVERED_BLOCK)) {
+            return;
+        }
         if (event.localNodeMaster() == false) {
             return;
         }
 
         Version minNodeVersion = event.state().getNodes().getMinNodeVersion();
         final List<UpdateAction> toRun = updateActions.stream()
             .filter(action -> action.isMinNodeVersionSupported(minNodeVersion))
+            .filter(action -> action.isAbleToRun(event.state()))
```

Comment:
Should we do this after we check for completed or currently_updating actions? Those `isAbleToRun` checks will be more expensive.

Reply:
Definitely, let me fix the filter ordering.